### PR TITLE
No space pdb title

### PR DESCRIPTION
**Pull request by rob-miller**
This pull request fixes an apparent bug in parse_pdb_header where a `space` is pre-pended to TITLE record data when read from a PDB file.  The original code adds the space to separate subsequent appended lines when processing multi-line titles, but no check is made to avoid appending the space to the initial `""` string.

I modified test_PDB to remove the initial space from the comparison lines in the header checks, and added a `test_parse_title()` check in test_PDB_parse_pdb_header which confirms that the three line title from 1LCD.pdb (already in Tests/PDB) is read in correctly.

This work also had the side effect of applying black formatting to the two test_PDB files mentioned.

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``flake8`` locally, and
understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [ ] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

**Comment by peterjc** [comment]
Have you run black on the test code as part of this? That would explain the large diff which makes reviewing your actual changes a little harder.

File: Tests/test_PDB.py [hunk 1]
```diff
@@ -741,89 +820,96 @@ def test_strict(self):
...
+            "O C O C N C C O C N C C O C S N C C O C C C O N "
+            "N C C O O O O O O O O O O O O O O O O O O O O O "
+            "O O O O O O O O O O O O O O O O O O O O O O O O "
+            "O O O O O O O O O O O O O O O O O O O O O O O O "
+            "O O O O O O O O O O O O O O O O O O O O O",
+        )
 
     def test_model_numbering(self):
         """Preserve model serial numbers during I/O."""
+
```
Comment (peterjc) [inline Tests/test_PDB.py]: This blank line is a problem:

```
$ flake8
./Tests/test_PDB.py:911:1: D202 No blank lines allowed after function docstring
```

For now we've putting a comment line here to avoid the ``D202`` versus ``black`` disagreement.

**Comment by rob-miller** [comment]
> Have you run black on the test code as part of this? That would explain the large diff which makes reviewing your actual changes a little harder.

Yes, sorry - I've set black as the formatter for my development and thought this would be a net positive.  Also sorry I overlooked the flake8 issue, it was highlighted at my end.

I've fixed the D202 problem, would you prefer I remove the black formatting from the test_ files or have your already gone through them now?  The only lines changed outside the black formatting in test_PDB are 158 and 176:

`"name": "hiv capsid c-terminal domain",`
`"name": "3d structure of alzheimer's abeta(1-42) fibrils",`

**Comment by rob-miller** [comment]
I've updated this PR to separate out the black formatting commits from the actual code changes.

File: Bio/PDB/parse_pdb_header.py [hunk 2]
```diff
@@ -214,7 +214,7 @@ def _parse_pdb_header_list(header):
         # From here, all the keys from the header are being parsed
         if key == "TITLE":
             name = _chop_end_codes(tail).lower()
```
Comment (JoaoRodrigues) [inline Bio/PDB/parse_pdb_header.py]: Ignoring the fact that we are inheriting a `dict`-named dictionary... I'd make use of the `dict.get` method here:
```
dict_name = dict.get('name', '')  # returns empty string if not found
dict_name = ' '.join([dict_name, name]).strip()  # faster than string concatenation
```

File: Tests/test_PDB_parse_pdb_header.py [hunk 3]
```diff
@@ -55,6 +87,16 @@ def test_parse_header_line(self):
         self.assertEqual(header["idcode"], "3EFG")
         self.assertEqual(header["deposition_date"], "2008-09-08")
 
+    def test_parse_title_line(self):
```
Comment (JoaoRodrigues) [inline Tests/test_PDB_parse_pdb_header.py]: I'd add a second test with an empty `header['name']` just to make sure it still returns something sensible.

File: Bio/PDB/parse_pdb_header.py (hunk 2 above)
Comment (rob-miller) [reply Bio/PDB/parse_pdb_header.py]: > Ignoring the fact that we are inheriting a `dict`-named dictionary... 

As we are in agreement here I have renamed to pdbh_dict

> I'd make use of the `dict.get` method here:
> 
> ```
> dict_name = dict.get('name', '') + ' ' + name
> dict_name = dict_name.strip()
> ```

just pushed a modified version incorporating this - with addition that pdbh_dict{"name"] is defined at the beginning (legacy) to be "" so can just do the (.join()).strip().

File: Tests/test_PDB_parse_pdb_header.py (hunk 3 above)
Comment (rob-miller) [reply Tests/test_PDB_parse_pdb_header.py]: done - I think this is your intent, unless you mean adding an additional test file with 'TITLE' in the header but nothing after it.

**Comment by rob-miller** [comment]
On the formatting, that's the result of black added to my development pipeline (so it runs every time I write a file out).   I could take it out but I think the goal is to black-ify all the code anyway?  My effort at separating the formatting out was to put it as separate commits at the beginning so you can see the effect separately.  I admit I have not scrutinized black results at all :-)

**Comment by peterjc** [comment]
@rob-miller would submitting a separate pull request just running black on ``Bio/PDB/`` first make life easier for you (#2008)? This PR should then merge and/or rebase nicely.

**Comment by rob-miller** [comment]
Thanks @peterjc, #2264 blackened Bio/PDB - these changes are in Tests/.  I can't find the comment I thought I saw about why not to apply black to Tests/ as you suggested initially, but went with the 'black changes in one commit' to try to separate them out here.

**Comment by peterjc** [comment]
@JoaoRodrigues are you able to merge pull requests? If not, do you want to be able to?

**Comment by peterjc** [comment]
Thanks Rob, merged.

**Comment by JoaoRodrigues** [comment]
Hi @peterjc , I am able to, but I wasn't that confident about it since I wasn't active for a long time. If that's OK with you, whenever we all agree that some contribution is good enough, I can merge it if it's assigned to me.

Thanks for merging and thanks @rob-miller for the contribution and the patience! Happy new year to all of you.

**Comment by peterjc** [comment]
@JoaoRodrigues that should be fine - better than us waiting for each other ;)
